src/gmail.js

```diff
@@ -961,6 +961,26 @@ var Gmail = function(localJQuery) {
             if (pathname && (pathname.endsWith("/i/s") || pathname.endsWith("/i/fd"))) {
                 api.tools.parse_request_payload(params, triggered);
             }
+            if (pathname && (pathname.endsWith("/i/s"))) {
+                if (params.body_params && params.body_params[2] && params.body_params[2][1] && params.body_params[2][1][0][2][2][7]) {
+                    // check params length  and get thread ids
+                    // console.log('params.body_params: ', params.body_params);
+                    if (params.body_params[2][1][0][2][2][7][2].length > 3) {
+                        const newThreadIds = params.body_params[2][1].map(obj => obj[2][1]);
+                        const newMessageIds = params.body_params[2][1].map(obj => obj[2][2][7][3]);
+                        // console.log(newMessageIds);
+                        // console.log('newThreadIds: ', newThreadIds);
+                        if (params.body_params[2][1][0][2][2][7][1][0] === "^k") {
+                            // console.log("delete_new");
+                            triggered["delete_new"] = [newThreadIds, newMessageIds];
```

Comment:
I don't think it makes sense to create events with *new names* just because it's a new version of Gmail underneath. We should just keep using `"delete"` and `"archive"` and the gmailjs library should abstract away this as much as possible.

For instance `"send_message"`-event has the same name for new and old data-layer.

Reply:
for this logic, we need mapping new threadIds with old threadIds

Reply:
Unfortunately, I think the documentation given so far says there's no straightforward or reliable way to do that.

In my extension, the hacky way I've done this is simply to *navigate* to the new thread/email-ID and pick up the legacy-ID from the DOM. But that's hardly an approach we can use inside an event-handler.

That said: I still think it would be bad to create *new* events just because the data layer changes, and would rather not merge a PR with such a change.

Maybe we can make the new event-signature `[null, params.url, params.body, new_email_ids, new_thread_ids]`. That would have a backward compatible signature, with a way of checking for the presence of the legacy-id.

Iirc there has been a few different cases in GmailJS where the email-ID cannot be directly inferred and its up to the extension-author to determine what is the best approach for resolving that after the fact (or maybe by using a stored emailID in a closure, etc).

Leaving the legacy value `null` will give extension-authors the same flexibility here, at least until we upgrade our data-APIs or find another workaround for this problem.

Reply:
ok, this is a cool idea with a new signature